src/main/java/com/intuit/graphql/orchestrator/utils/MultiEIGenerator.java

```diff
@@ -39,7 +43,12 @@ public Flux<ExecutionInput> generateEIs() {
                 this.timeProcessedSplit = System.currentTimeMillis();
                 //Adds elements to list of eis that need to be processed
                 try {
-                    this.eis.addAll(MultipartUtil.splitMultipartExecutionInput(emittedEI));
+                    Document rootDocument = new Parser().parseDocument(emittedEI.getQuery());
```

Comment:
I think Parser object is thread-safe.  We can have a static one.

Reply:
Also, lets see if we already have a static parser somewhere in the code. Same for AstTransformer

Reply:
Located in GraphQLUtil file.. Next commit will have changes